### PR TITLE
doc: release: 3.7: Add release notes for Flash

### PR DESCRIPTION
**Pull request by de-nordic**
Flash API additions and new drivers.

### CODE REVIEW

File: doc/releases/release-notes-3.7.rst [hunk 1]
```diff
@@ -377,6 +377,18 @@ Drivers and Sensors
 * Flash
 
   * Added support for Ambiq Apollo3 series.
+  * Added support for multi-instance of SPI NOR driver (spi_nor.c).
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:380]:
```suggestion
  * Added support for multiple instances of the SPI NOR driver (spi_nor.c).
```
maybe better?

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -377,6 +377,18 @@ Drivers and Sensors
...
+  * Added preliminary support for non-erase devices with introduction of
+    device capabilities to c:struct:`flash_parameters` and utility function
+    c:func:`flash_params_get_erase_cap` that allows to obtain erase type
+    provided by a device; added c:macro:`FLASH_ERASE_C_EXPLICIT`, which is
+    currently only supported erase type and is set by all flash devices.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:385]:
```suggestion
    device capabilities to c:struct:`flash_parameters` and the utility function
    c:func:`flash_params_get_erase_cap` that allows to obtain the erase type
    provided by a device; added c:macro:`FLASH_ERASE_C_EXPLICIT`, which is
    currently the only supported erase type and is set by all flash devices.
```
maybe better?

File: doc/releases/release-notes-3.7.rst [hunk 3]
```diff
@@ -377,6 +377,18 @@ Drivers and Sensors
...
+  * Added c:func:`flash_flatten` function that can be used on devices
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:386]:
```suggestion
  * Added the c:func:`flash_flatten` function that can be used on devices
```

File: doc/releases/release-notes-3.7.rst [hunk 4]
```diff
@@ -377,6 +377,18 @@ Drivers and Sensors
...
+    with and without erase when erase is used not for preparing device for
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:387]: If there something missing before that "when"?
(I can't really parse this sentence)

File: doc/releases/release-notes-3.7.rst [hunk 5]
```diff
@@ -377,6 +377,18 @@ Drivers and Sensors
...
+    random data write but rather to remove/scramble device.
+  * Added c:func:`flash_fill` utility function that allows to write
+    single value across provided range of selected device.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:390]:
```suggestion
  * Added the c:func:`flash_fill` utility function which allows to write
    a single value across a provided range in a selected device.
```

**Review by aescolar — COMMENTED** [review]
Thanks @de-nordic, some minor requests

File: doc/releases/release-notes-3.7.rst (hunk 4 above)
Comment (de-nordic) [reply doc/releases/release-notes-3.7.rst]: Improved, hopefully, the description.

**Review by nashif — APPROVED** [review]

**Review by aescolar — APPROVED** [review]

**Pull request merged**